### ISSUE TITLE
Lumo Customisation & CSS Specificity

### ISSUE THREAD

**Issue by anezthes** [vaadin/vaadin-themable-mixin#28]
_Talked with Virkki, Koivuviita and Korolkov and they said I should report this issue here. Mentioned they already have a solution in place._

**Issue:**
If I want to change the font-family for (a Lumo themed) vaadin-grid, I have to write:

```
<dom-module id="my-theme" theme-for="vaadin-grid">
  <template>
    <style>
      :host(.my-theme) {
        font-family: Arial;
      }
    </style>
  </template>
</dom-module>
```

instead of

```
<dom-module id="my-theme" theme-for="vaadin-grid">
  <template>
    <style>
      :host {
        font-family: Arial;
      }
    </style>
  </template>
</dom-module>
```

**Comment by jouni** [comment]
Adding a bit more context: the issue is that in some cases (and apparently sometimes randomly) the order in which the style modules are applied is so that app-specific modules are applied before the built-in theme module.

The workaround is to use a higher specificity selector in the app-specific module (or `!important`, but that’s discouraged).

A partial fix is that we mark the built-in theme modules somehow and ThemableMixin then always applies those first in the DOM. App-specific modules would still be loaded in random order.

**Comment by jouni** [comment]
Also, this behavior should be documented, that ThemableMixin does not guarantee the order in which the style modules are applied.

**Comment by Legioth** [comment]
I believe it would be a good solution to by default include styles from built-in theme modules before custom ones. This could be implemented so that `ThemableMixin` looks for a `priority` attribute on each module and includes styles in priority order. If no priority is defined, `0` would be implied. Built-in modules would all have `-1` set as their priority so that they would be loaded before modules without a priority (and thus allow modules without a priority to override built-in rules with the same selector specificity).

To further help make the process more deterministic, I would suggest ordering modules with the same priority based on their `id` value.

**Comment by web-padawan** [comment]
Appears to be the same as #38 and that was addressed by #43 and #44
